probe/tag/tagger.go

```diff
@@ -0,0 +1,25 @@
+package tag
+
+import "github.com/weaveworks/scope/report"
+
+// Tagger tags nodes with value-add node metadata.
+type Tagger interface {
+	Tag(r report.Report, ts report.TopologySelector, id string) report.NodeMetadata
```

Comment:
> Do we ever tag anything other than a whole report?

No.

> Is there any case where we might want to apply the same tagger to different topologies?

Yes. In fact, we almost always want to apply the same tagger to multiple topologies. For example, the docker tagger should tag nodes in any topology that can be linked to a process, and that includes Endpoints, Addresses, Containers, Processes, and potentially more that we add in the future.

— – -

Maybe we can skip a little discussion — are you driving at something like

```go
type Tagger interface {
    Tag(Report)
}
```

?